### DISCUSSION TITLE
Local image build with `ko`

### DISCUSSION THREAD

**Discussion by grzesuav** [goreleaser/goreleaser#4678] [Q&A] [answered]
Hi,

can I somehow trigger `ko` builds locally ? In docs I see
```
Ko only runs on the publishing phase, so it might be a bit hard to test — you might need to push to a fake repository (or a fake tag) when playing around with its configuration.
```
however I wanted to builds image the same way for local development and ci testing, so to build and load `ko` images to local registry, from what I can read in https://ko.build/configuration/#local-publishing-options `ko` can do it :
```
ko can also load images to a local Docker daemon, if available, by setting KO_DOCKER_REPO=ko.local, or by passing the --local (-L) flag.
```
however unsure f it can be achieved when running `ko` from `goreleaser`

**Comment by grzesuav** [comment]
Two additional quesitons for `ko`:
1. How to set docker image labels ?
2. Is it possible to add docker image url to release notes? To be visible on somewhere near changelog/artifacts ?

**Reply by grzesuav** [reply]
1. I am blind, this is one of the ko configuration option

**Reply by caarlos0** [reply]
for 2 you can template using `{{ .Artifacts }}` (only on pro, though) 

https://goreleaser.com/customization/templates/#common-fields

**Reply by grzesuav** [reply]
Got it, for me the footer could be a solution if I get it working, thanks

**Comment by caarlos0** [comment]
I think currently ko will not run on `--snapshot`, so it's not possible to test it locally... 

maybe worth opening an issue for it though, we could see if it's snapshot and automatically set `KO_BASE_REPO` to local 🤔

**Reply by grzesuav** [reply]
Issue against this repository or KO?

**Reply by caarlos0** [reply]
on goreleaser's repo

**Reply by grzesuav** [reply, marked as answer]
https://github.com/goreleaser/goreleaser/issues/4683
